app/livechat/server/lib/Livechat.js

```diff
@@ -412,8 +412,10 @@ export const Livechat = {
 			}
 		}
 
-		if (customField.scope === 'room') {
-			return LivechatRooms.updateDataByToken(token, key, value, overwrite);
+		if (!hasPermission()) {
```

Comment:
Is this check correct? what is this test supposed to do?
Also, remember that non-authenticated users will call this method(`Livechat.setCustomFields`) like the Livechat visitors, so I don't think we can check permissions for this case.